src/test/java/tests/JavaVersionTest.java

```diff
@@ -0,0 +1,135 @@
+package tests;
+
+import org.junit.Test;
+import util.JavaVersion;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.fail;
+
+public class JavaVersionTest {
+    @Test
+    public void javaVersionParsing_Java8VersionNoBuild_ParsedVersionCorrectly() {
```

Comment:
Not for now, but there is a proposed change for Java 9 where the version format will change. Should I remove `8` for now?